update external check docs with deadline (#452)
Comcast/kuberhealthy: docs/EXTERNAL_CHECK_CREATION.md

- Read the `KH_REPORTING_URL` environment variable.
- Send a `POST` to the `KH_REPORTING_URL` with the following JSON body:

> - Ensure that your check finishes within the `unix` deadline specified in the `KH_CHECK_RUN_DEADLINE` environment variable. Without respecting `KH_CHECK_RUN_DEADLINE`, your check may run into a `400` error returned from Kuberhealthy when reporting your check's status. 

Reviewer: ```suggestion
- Ensure that your check finishes within the [unixtime](https://en.wikipedia.org/wiki/Unix_time) deadline specified in the `KH_CHECK_RUN_DEADLINE` environment variable. If `KH_CHECK_RUN_DEADLINE` is not rexpected, your check may run into a `400` error when reporting state to Kuberhealthy. 
```

Clients outside of Go can be found in the [clients directory](../clients).

#### Injected Check Pod Environment Variables

> 

Reviewer: ```suggestion
The following environment variables are injected into every checker pod that Kuberhealthy runs.  When writing your checker code, you can depend on these environment variables always being available to you, even if you do not specify them in your `khcheck` spec..
```

Did we get all of them?  Can you check that environment variable injecting code you deduplicated a while ago and verify please?

Author: These are not all of them -- but these are the ones that checks should be aware of.

The other ones are `UUID`, `external check POST URL`, etc.
